### PR TITLE
USHIFT-1097 Test upgrade from Engineering Candidate

### PR DESCRIPTION
**Pull request by pmtk**

### CODE REVIEW

File: test/scenarios/rhel-9.2-microshift-source-from-latest-ec.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (pmtk) [inline test/scenarios/rhel-9.2-microshift-source-from-latest-ec.sh:1]: Planning to move the file to `scenarios-periodics` after https://github.com/openshift/microshift/pull/2149/ merges.
@dhellmann @pacevedom I guess we'll run all periodics in a single job?

Comment (dhellmann) [reply test/scenarios/rhel-9.2-microshift-source-from-latest-ec.sh:1]: Yes, that's what I had in mind. I think @pacevedom 's work will make that easy.

File: test/image-blueprints/group2/rhel92-microshift-latest-ec.toml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+description = "RHEL 9.2 with latest MicroShift Engineering Candidate."
+version = "0.0.1"
+modules = []
+groups = []
+distro = "rhel-92"
+
+# FIXME: Temporarily pin the openvswitch version.
+[[packages]]
+name = "openvswitch3.1"
+version = "3.1.0-14.el9fdp"
```
Comment (dhellmann) [inline test/image-blueprints/group2/rhel92-microshift-latest-ec.toml:11]: We can remove this pin for 4.14 ECs now.

File: test/package-sources/microshift-latest-ec.toml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+id = "microshift-latest-ec"
+name = "Latest MicroShift Engineering Candidate"
+type = "yum-baseurl"
+url = "https://mirror.openshift.com/pub/openshift-v4/${UNAME_M}/microshift/ocp-dev-preview/latest-4.14/elrhel-9/os/"
```
Comment (dhellmann) [inline test/package-sources/microshift-latest-ec.toml:4]: Just FYI, we _might_ need to change the `elrhel-9` portion of this path back to just `el9` later. @vfreex was going to look at fixing the pipeline change that introduced that path change for EC4.

**Comment by pmtk** [comment]
/unhold

Changed composer's source to use $MINOR_VERSION in the url:
```
url = "https://mirror.openshift.com/pub/openshift-v4/${UNAME_M}/microshift/ocp-dev-preview/latest-4.${MINOR_VERSION}/elrhel-9/os/"
```

**Comment by pmtk** [comment]
The test currently fails due to recent change of `version` schema (from `X.Y.Z` to json):
```
failed to get version of existing MicroShift data: failed to read version file: unmarshalling \"4.14.0\" failed: invalid character '.' after top-level value
```

@dhellmann is it okay that we don't implement "migration" from one schema to the other and just wait for new EC/RC?

**Comment by pmtk** [comment]
Let's wait for https://github.com/openshift/release/pull/41823 so we can test this as presubmit
/hold

**Comment by dhellmann** [comment]
> The test currently fails due to recent change of `version` schema (from `X.Y.Z` to json):
> 
> ```
> failed to get version of existing MicroShift data: failed to read version file: unmarshalling \"4.14.0\" failed: invalid character '.' after top-level value
> ```
> 
> @dhellmann is it okay that we don't implement "migration" from one schema to the other and just wait for new EC/RC?

I think right now that's safe. After we have a GA release we would have to deal with format conversions like that.

If we don't make the change now, how long will it be before we have a candidate release that will pass the job with the right data format?

**Comment by pmtk** [comment]
I'm planning to revisit this next sprint with some dynamic logic to pick EC, RC, or even released RPMs.

**Pull request closed without merging**